lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php

```diff
@@ -36,6 +39,10 @@
  */
 abstract class AbstractMySQLDriver implements Driver, ExceptionConverterDriver, VersionAwarePlatformDriver
 {
+
+    private const MYSQL_MARIADB_VERSION_REGEXP = '/^(mariadb-)?(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)/';
```

Comment:
Is there a reason to allow only one digit in the MySQL version? E.g. what are the supported versions which contain only one digit? According to the [documentation](https://dev.mysql.com/doc/refman/5.5/en/which-version.html), there are always 3 digits:
> The naming scheme in MySQL 5.5 uses release names that consist of three numbers and an optional suffix; for example, **mysql-5.5.1-m2**.

Reply:
This value is reused neither by the class itself nor outside. Why make it a constant?

Reply:
@morozov:

Answer to both questions can be found previous reviews and directions ;)

> This value is reused neither by the class itself nor outside. Why make it a constant?
 :+1: 1  

Preview review by lcobucci: `We can extract the regex into a private const to make things more readable.` See https://github.com/doctrine/dbal/pull/2825/commits/fc0f2e3537771e4feaee6aec69eacefee1ef0b6a?diff=split

I'll revert that one. no offense lcobucci ;)  

> Is there a reason to allow only one digit in the MySQL version? 

From the firection I had before: +/- don't bc-break. It's explicitly defined in tests (both 1 and two numbers): 
https://github.com/doctrine/dbal/blob/9d706672787826e6f160bc5c3ccada6e42b946af/tests/Doctrine/Tests/DBAL/Driver/AbstractMySQLDriverTest.php#L59 and https://github.com/doctrine/dbal/blob/9d706672787826e6f160bc5c3ccada6e42b946af/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php#L127

The question now, is that if it can introduce a bc-break ?

For the 'one' number question, intersting to see that mysql v6 was covered (they skip the version to 8)
https://github.com/doctrine/dbal/blob/9d706672787826e6f160bc5c3ccada6e42b946af/tests/Doctrine/Tests/DBAL/Driver/AbstractMySQLDriverTest.php#L64. 

I'm less sure about this:

```
     $majorVersion = $versionParts['major'];
        $minorVersion = isset($versionParts['minor']) ? $versionParts['minor'] : 0;
        $patchVersion = isset($versionParts['patch']) ? $versionParts['patch'] : null;
        if ('5' === $majorVersion && '7' === $minorVersion && null === $patchVersion) {
            $patchVersion = '9';
        }
        $version = $majorVersion . '.' . $minorVersion . '.' . $patchVersion;
        if (version_compare($version, '5.7.9', '>=')) {
            return new MySQL57Platform();
        }
```

@morozov As it iss a question I asked, explained before and coming up again and again. I'll add a new comment in the discussion ;) For now I don't change anything, but I'll summarize

Reply:
> I'll revert that one. no offense lcobucci ;)

None taken 😄 

Regardless of how it's used I tend to give proper names to things so it's easier to understand what I'm trying to achieve, it was merely a suggestion and it's all good.